diff -r -U 3 minecraft_server\net\minecraft\entity\passive\EntityHorse.java minecraft_server_patched\net\minecraft\entity\passive\EntityHorse.java
--- minecraft_server\net\minecraft\entity\passive\EntityHorse.java
+++ minecraft_server_patched\net\minecraft\entity\passive\EntityHorse.java
@@ -122,7 +122,7 @@
       this.field_70180_af.func_187214_a(field_184787_bE, Byte.valueOf((byte)0));
       this.field_70180_af.func_187214_a(field_184788_bF, Integer.valueOf(HorseArmorType.HORSE.func_188595_k()));
       this.field_70180_af.func_187214_a(field_184789_bG, Integer.valueOf(0));
-      this.field_70180_af.func_187214_a(field_184790_bH, Optional.<T>absent());
+      this.field_70180_af.func_187214_a(field_184790_bH, Optional.<UUID>absent());
       this.field_70180_af.func_187214_a(field_184791_bI, Integer.valueOf(HorseType.NONE.func_188579_a()));
    }
 
@@ -179,7 +179,7 @@
    }
 
    public void func_184779_b(UUID p_184779_1_) {
-      this.field_70180_af.func_187227_b(field_184790_bH, Optional.<T>fromNullable(p_184779_1_));
+      this.field_70180_af.func_187227_b(field_184790_bH, Optional.fromNullable(p_184779_1_));
    }
 
    public float func_110254_bY() {
