diff -r -U 3 minecraft\net\minecraft\block\properties\PropertyBool.java minecraft_patched\net\minecraft\block\properties\PropertyBool.java
--- minecraft\net\minecraft\block\properties\PropertyBool.java
+++ minecraft_patched\net\minecraft\block\properties\PropertyBool.java
@@ -28,7 +28,7 @@
 
    @SideOnly(Side.CLIENT)
    public Optional<Boolean> func_185929_b(String p_185929_1_) {
-      return !"true".equals(p_185929_1_) && !"false".equals(p_185929_1_)?Optional.absent():Optional.of(Boolean.valueOf(p_185929_1_));
+      return !"true".equals(p_185929_1_) && !"false".equals(p_185929_1_)?Optional.<Boolean>absent():Optional.of(Boolean.valueOf(p_185929_1_));
    }
 
    public boolean equals(Object p_equals_1_) {
